src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        'The side of the figure cannot be less than or equal to zero',
+      );
+    }
+
+    const maxLength: number = Math.max(a, b, c);
+    const totalLength: number = a + b + c;
+
+    if (maxLength >= totalLength - maxLength) {
+      throw new Error("Sides 1, 2, and 3 can't form a triangle");
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter: number = 0.5 * (this.a + this.b + this.c);
+    const area: number = Math.sqrt(
+      semiperimeter *
+        (semiperimeter - this.a) *
+        (semiperimeter - this.b) *
+        (semiperimeter - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - The same applies here as well. TypeScript implicitly creates properties on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to do it manually.